blocks/form/form.js

```diff
@@ -0,0 +1,182 @@
+      if (fe.checked) payload[fe.id] = fe.value;
+    } else if (fe.id) {
+      payload[fe.id] = fe.value;
+    }
+  });
+  return payload;
+}
+
+async function submitForm(form) {
+  const payload = constructPayload(form);
+  const resp = await fetch(form.dataset.action, {
+    method: 'POST',
+    cache: 'no-cache',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ data: payload }),
+  });
+  await resp.text();
+  return payload;
+}
+
+function createButton(fd) {
+  const button = document.createElement('button');
+  button.textContent = fd.Label;
+  button.classList.add('button');
+  if (fd.Type === 'submit') {
+    button.addEventListener('click', async (event) => {
+      const form = button.closest('form');
+      if (form.checkValidity()) {
+        event.preventDefault();
+        button.setAttribute('disabled', '');
+        await submitForm(form);
+        const redirectTo = fd.Extra;
+        window.location.href = redirectTo;
+      }
+    });
+  }
+  return button;
+}
+
+function createHeading(fd) {
+  const heading = document.createElement('h3');
+  heading.textContent = fd.Label;
+  return heading;
+}
+
+function createInput(fd) {
+  const input = document.createElement('input');
+  input.type = fd.Type;
+  input.id = fd.Field;
+  input.setAttribute('placeholder', fd.Placeholder);
+  if (fd.Mandatory === 'x') {
+    input.setAttribute('required', 'required');
+  }
+  return input;
+}
+
+function createTextArea(fd) {
+  const input = document.createElement('textarea');
+  input.id = fd.Field;
+  input.setAttribute('placeholder', fd.Placeholder);
+  if (fd.Mandatory === 'x') {
+    input.setAttribute('required', 'required');
+  }
+  return input;
+}
+
+function createLabel(fd) {
+  const label = document.createElement('label');
+  label.setAttribute('for', fd.Field);
+  label.textContent = fd.Label;
+  if (fd.Mandatory === 'x') {
+    label.classList.add('required');
+  }
+  return label;
+}
+
+function applyRules(form, rules) {
+  const payload = constructPayload(form);
+  rules.forEach((field) => {
+    const { type, condition: { key, operator, value } } = field.rule;
+    if (type === 'visible') {
+      if (operator === 'eq') {
+        if (payload[key] === value) {
+          form.querySelector(`.${field.fieldId}`).classList.remove('hidden');
+        } else {
+          form.querySelector(`.${field.fieldId}`).classList.add('hidden');
+        }
+      }
+    }
+  });
+}
+
+async function createForm(formURL) {
+  const { pathname } = new URL(formURL);
+  const resp = await fetch(pathname);
+  const json = await resp.json();
+  const form = document.createElement('form');
+  const rules = [];
+  // eslint-disable-next-line prefer-destructuring
+  form.dataset.action = pathname.split('.json')[0];
+  json.data.forEach((fd) => {
+    fd.Type = fd.Type || 'text';
+    const fieldWrapper = document.createElement('div');
+    const style = fd.Style ? ` form-${fd.Style}` : '';
```

Comment:
What would this field contain? Having style definitions in the content is a 🚩

Reply:
This is the original form block in helix-website. I thought that is a pattern that is being followed. We are not using that for nedbank.

Reply:
Ok, then let's take it out for now. We can fix the `helix-website` later.

Reply:
Removed it